### ISSUE TITLE
Create and implement a Submit PSA/Event Form

### ISSUE THREAD

**Issue by BigActual** [Good-Shepherd-Catholic-Radio/Theme-2017#30]

**Comment by d4mation** [comment]
Would this be for Visitors to add their own Events to the Calendar?

If so, this should handle it for us and tie into The Events Calendar nicely.

https://theeventscalendar.com/product/wordpress-community-events/

**Comment by BigActual** [comment]
![bitmoji](https://render.bitstrips.com/v2/cpanel/7681590-283868566_2-s4-v1.png?transparent=1&palette=1&width=246)

**Comment by BigActual** [comment]
This form is present... Ugh it's long though. This requires tie into the Home Page interface. I don't seem to have controls here for adding forms to this section.

**Comment by BigActual** [comment]
Also we need to tie this into an approval workflow so that an approved PSA becomes an event. Now that I'm thinking about this, I think I botched the form. Should this not have been a Gravity Form? Was there another tool to use for submitting EventCalendar Pro Events for approval?

**Comment by d4mation** [comment]
https://theeventscalendar.com/product/wordpress-community-events/

**Comment by BigActual** [comment]
[PSA Policies and Guidelines Revised.pdf](https://github.com/Good-Shepherd-Catholic-Radio/Theme-2017/files/1279884/PSA.Policies.and.Guidelines.Revised.pdf)
> The "Events/PSA Form Will Go Here" section should be titled "Events/Public Service Announcement Policies and Forms" with links to the attached "Policies and Guidelines" and "Forms".

> Thank you for your help!

> Harry Hill

**Comment by BigActual** [comment]
Are you planning on implementing this on Tuesday?

**Comment by d4mation** [comment]
I can implement it before, but I need a copy of the plugin necessary. I know the license was bought, but I do not have the actual .ZIP file.

**Comment by BigActual** [comment]
OK, let me ask Harry for it.

**Comment by BigActual** [comment]
Followed up with a phone call

**Comment by BigActual** [comment]
[the-events-calendar-community-events.4.5.5.zip](https://github.com/Good-Shepherd-Catholic-Radio/Theme-2017/files/1293530/the-events-calendar-community-events.4.5.5.zip)

**Comment by d4mation** [comment]
Public Repo, so deleted that bit

**Comment by BigActual** [comment]
[the-events-calendar.4.5.12.2.zip](https://github.com/Good-Shepherd-Catholic-Radio/Theme-2017/files/1293551/the-events-calendar.4.5.12.2.zip)
[event-tickets.4.5.5.zip](https://github.com/Good-Shepherd-Catholic-Radio/Theme-2017/files/1293553/event-tickets.4.5.5.zip)

**Comment by d4mation** [comment]
Made the "Events/PSA Form goes here" bit into something that can be edited within WordPress.

The current title is a bit long though. We will either want to shorten it somehow, or change the font-size.

![image](https://user-images.githubusercontent.com/7770631/30291770-e1d9e550-9701-11e7-97cf-d3e4c34945bc.png)

**Comment by d4mation** [comment]
@BigActual 

Been messing with this for a few hours now. The AJAX Live-Refreshing is not playing well with the form this makes at all. I even forced every Script that would be added to the page to get loaded one after another Synchronously (your Web Browser normally does _not_ allow this. I had to get very creative to force it. This is also not very efficient and slower.) and it **still** causes JavaScript errors stating that things which previously were loaded are inaccessible (Even basic stuff like jQuery, which is _always_ loaded)

However, if we were to disable Live Refreshing _just for the Event/PSA Submissions Process_, we wouldn't need to worry about this. This could be a stop-gap potentially.

Just wanted to throw that out as an option. I don't know if/when I could get this worked out. Getting the Events Calendar itself to work properly on Live Refresh was a Herculean feat and this is proving to be the same.

In a similar vein, this Form _cannot_ be in an Overlay. This is because I cannot force it to load its necessary scripts as I want to. If I could tell it "Hey, load your scripts always. I don't mind." we wouldn't be having these problems, but unfortunately their code doesn't check if it should run (Instead of asking "Does that Date Picker field exist?" before creating it, it just always tries to create it) which causes tons of JavaScript errors.

**Comment by d4mation** [comment]
Took a lot of tweaking to ensure members of the Community couldn't create Radio Shows, but I now have this set on the server _without_ Ajax.

I explicitly set any Community Events page to not load using Ajax, but if we ever need to for another reason, adding the class `no-ajax` to a link will also prevent that page from loading via Ajax.

I'm going to label this as "Needs Ajax" and leave this open so that we may be able to come back to it at some point.

**Comment by d4mation** [comment]
A thought that I just had with this:

Something that we could add in the meantime would be that if a Page cannot be loaded via AJAX for one reason or another (Such as this one), if they're actually listening to the Radio Stream, it could throw up a confirmation dialog saying "Visiting this page will stop the Radio Stream. Are you sure?" and if they select "yes", they will be taken to the new page. 

Not a perfect solution, but a good stop-gap. The Ajax Live-Refreshing is only truly there to benefit the Radio Stream. If they aren't listening to it, they may not even notice that the page did not Live-Refresh.
